src/testing/jest/README.md

```diff
@@ -0,0 +1,210 @@
+# Jest Support in Stencil
+
+This directory contains support for the Jest testing library.
+
+## Adding Support for a New Version of Jest
+
+The steps for adding support for a new version of Jest can be found below.
+These steps are intended to be followed in order, and be a part of a single pull request.
+**It is HIGHLY recommended that you create a commit for each step in the process.**
+This allows for 'well known' commits that map directly back to these steps to be created, speeding up the review process.
+
+1. In this directory, there are several version-specific directories to support Jest.
+   For example, `jest-27-and-under/` is for Jest v27 and below, `jest-28/` is for Jest v28, etc.
+   To start, create a copy of the entire directory contents for the latest version of Jest in this directory, ensuring the version number is in the directory name.
+   Note that `jest-27-and-under/` is a special case of supporting more than one version of Jest.
+   Going forward, there should be one directory for each version of Jest that is supported.
+   
+   If we wanted to add support for Jest v28 and had the following directory structure:
+   ```
+   └── src/testing/jest/
+       └── jest-27-and-under
+   ```
+   Adding support for Jest v28 would begin with making a copy of `jest-27-and-under/` (the latest supported version) and naming it `jest-28/`.
```

Comment:
Can we somehow make the "latest supported version" callout a bit more prominent? I think we all have enough intuition to use the latest version's adapter in practice, but better safe than sorry. Only calling this out so we don't end up cloning the `jest-27-and-under` directory as the starting point each time since that has some special cases in files like the preprocessor that have to make additional version checks.

Reply:
👍 I tried to do this in https://github.com/ionic-team/stencil/pull/4952/commits/ac9c04aa148b8f70357cba91065077c9aff4dbda - I added Jest 28 into the example as if it already existed and the example goes through adding Jest 29. Hopefully that and some bolding of certain areas brings out the intent more

Reply:
Yep, think its pretty clear now. Changing the examples to be a migration from Jest 28 to 29 was a good move 👍